src/Services/RabbitMQService.cs

```diff
@@ -1,30 +1,44 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
 using RabbitMQ.Client;
 
 namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ
 {
     internal sealed class RabbitMQService : IRabbitMQService
     {
+        private IModel _channel;
         private IBasicPublishBatch _batch;
 
-        public RabbitMQService(string hostname, string queuename)
+        public RabbitMQService(string hostName, string queueName)
         {
-            _batch = CreateBatch(hostname, queuename);
+            string host = hostName ?? throw new ArgumentNullException(nameof(hostName));
+            string queue = queueName ?? throw new ArgumentNullException(nameof(queueName));
+            CreateChannel(host);
+            CreateBatch(queue);

```

Comment:
Add private variables for hostName and queueName. Move logic from CreateChannel and CreateBatch to constructor as those methods are not being used any where else